android-todo-list/architecture-samples-main/app/src/main/java/com/statsig/todoapp/data/DefaultTaskRepository.kt

```diff
@@ -83,10 +85,42 @@ class DefaultTaskRepository @Inject constructor(
         }
     }
 
-    override fun getTasksStream(): Flow<List<Task>> {
-        return localDataSource.observeAll().map { tasks ->
-            withContext(dispatcher) {
-                tasks.toExternal()
+    override fun getTasksStream(sortOrderValue: Int): Flow<List<Task>> {
+        Statsig.logEvent(
+            "${StatsigUtil.ITEM_SORT} ",
+            "${StatsigUtil.SORT_ORDER} $sortOrderValue"
+        )
+        return when (sortOrderValue) {
+            1 -> {
+                localDataSource.orderAlphabetically().map { tasks ->
+                    withContext(dispatcher) {
```

Comment:
The mapping function here is the same for all these cases. Just get the data and then map it.

```
val data = when (order) {
 TaskSortOrder.Alphabetical -> localDataSource.orderAlphabetically()
 TaskSortOrder.NewestFirst -> localDataSource.orderNewestDateWise()
 ...
} 

return data.map { tasks ->
    withContext(dispatcher) {
        tasks.toExternal()
    }
}
```

Reply:
Implemented.